src/main/java/com/puppycrawl/tools/checkstyle/xpath/SuppressionsFileGenerator.java

```diff
@@ -0,0 +1,176 @@
+////////////////////////////////////////////////////////////////////////////////
+// checkstyle: Checks Java source code for adherence to a set of rules.
+// Copyright (C) 2001-2018 the original author or authors.
+//
+// This library is free software; you can redistribute it and/or
+// modify it under the terms of the GNU Lesser General Public
+// License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+//
+// This library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+// Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public
+// License along with this library; if not, write to the Free Software
+// Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+////////////////////////////////////////////////////////////////////////////////
+
+package com.puppycrawl.tools.checkstyle.xpath;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.io.OutputStreamWriter;
+import java.io.PrintWriter;
+import java.nio.charset.StandardCharsets;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import com.puppycrawl.tools.checkstyle.JavaParser;
+import com.puppycrawl.tools.checkstyle.api.AuditEvent;
+import com.puppycrawl.tools.checkstyle.api.AuditListener;
+import com.puppycrawl.tools.checkstyle.api.AutomaticBean;
+import com.puppycrawl.tools.checkstyle.api.CheckstyleException;
+import com.puppycrawl.tools.checkstyle.api.DetailAST;
+import com.puppycrawl.tools.checkstyle.api.FileText;
+import com.puppycrawl.tools.checkstyle.api.LocalizedMessage;
+
+/**
+ * Creates <b>suppressions.xml</b> file, based on violations occurred.
+ *
+ * @author Timur Tibeyev.
+ */
+public class SuppressionsFileGenerator extends AutomaticBean implements AuditListener {
+
+    /**
+     * Helper writer that allows easy encoding and printing.
+     */
+    private final PrintWriter writer;
+
+    /** The distance between tab stop position. */
+    private final int tabWidth;
+
+    /** The delimiter between xpath queries. */
+    private static final String DELIMITER = " | \n";
+
+    private final String[] supportedChecks = {
+            "CyclomaticComplexity", "DeclarationOrder", "DefaultComesLast",
+            "ExplicitInitialization", "FallThrough", "HiddenField", "IllegalThrows",
+            "ImportControl", "JavadocVariable", "LeftCurly", "MethodParamPad",
+            "MultipleVariableDeclarations", "NestedForDepth", "NestedIfDepth", "NestedTryDepth",
+            "NPathComplexity", "OneStatementPerLine", "OuterTypeNumber", "RequireThis", "RightCurly"
```

Comment:
it is ok, but might forget to update this list.
What is we will keep this list in Filter  and have static method to verify if Check is supported ? or smth like this